de.peeeq.wurstscript/src/main/antlr/de/peeeq/wurstscript/antlr/Jass.g4

```diff
@@ -0,0 +1,206 @@
+grammar Jass;
+
+@header {
+    package de.peeeq.wurstscript.antlr;
+}
+
+compilationUnit : NL? decls+=jassTopLevelDeclaration*;
+
+jassTopLevelDeclaration:
+     jassGlobalsBlock
+   | jassFuncDef
+   | jassTypeDecl
+   | jassNativeDecl
+;
+
+jassGlobalsBlock:
+    'globals' NL
+        vars+=jassGlobalDecl*
+    'endglobals' NL
+;
+
+jassGlobalDecl:
+    constant='constant'? jassTypeExpr name=ID ('=' initial=jassExpr)? NL
+;
+
+jassFuncDef:
+    constant='constant'? 'function' jassFuncSignature NL
+        (jassLocals+=jassLocal)*
+        jassStatements
+    'endfunction' NL
+;
+
+jassLocal:
+    'local' jassTypeExpr name=ID ('=' initial=jassExpr)? NL;
+
+jassStatements:
+    stmts+=jassStatement*;
+
+jassStatement:
+      jassStatementIf
+    | jassStatementLoop
+    | jassStatementExithwhen
+    | jassStatementReturn
+    | jassStatementSet
+    | jassStatementCall
+    ;
+
+jassStatementIf:
+    'if' cond=jassExpr 'then' NL
+        thenStatements=jassStatements
+    jassElseIfs
+;
+
+jassElseIfs:
+      'elseif' cond=jassExpr 'then' NL
+            thenStatements=jassStatements
+        jassElseIfs
+    | 'else' NL
+        elseStmts=jassStatements
+      'endif' NL
+    | 'endif' NL
+;
+
+jassStatementLoop:
+    'loop' NL
+        jassStatements
+    'endloop' NL
+;
+
+jassStatementExithwhen: 'exitwhen' cond=jassExpr NL;
+
+jassStatementReturn: 'return' jassExpr NL;
+
+jassStatementSet: 'set' left=exprVarAccess '=' right=jassExpr NL;
+
+jassStatementCall:'call' exprFunctionCall NL;
+
+jassNativeDecl: constant='constant'? 'native' jassFuncSignature NL;
+
+jassFuncSignature:
+    name=ID 'takes' ('nothing'|args+=formalParam (',' args+=formalParam)*) 'returns' ('nothing'|returnType=jassTypeExpr)
+;
+
+jassTypeDecl: 'type' name=ID 'extends' extended=jassTypeExpr NL;
+
+formalParam: jassTypeExpr name=ID;
+
+jassTypeExpr:
+    | typeName=ID
+    | jassTypeExpr 'array'
+;
+
+exprVarAccess: varname=ID indexes?;
+
+indexes: '[' jassExpr ']';
+
+jassExpr:
+      jassExprPrimary
+    | left=jassExpr op=('*'|'/'|'%') right=jassExpr
```

Comment:
`%` is not valid in Jass

Reply:
The module operator `%` has been added with patch 1.29 according to @lep